narwhal/primary/src/synchronizer.rs

```diff
@@ -156,19 +181,6 @@ impl Inner {
 
         let digest = certificate.digest();
 
-        // Validate that certificates are accepted in causal order.
-        // This should be relatively cheap because of certificate store caching.
-        if certificate.round() > self.gc_round.load(Ordering::Acquire) + 1 {
```

Comment:
That check saved the day a couple of times during refactoring which made simtests fail - do we see high performance penalty leaving it?

Reply:
After the tokio Mutex fix, I think we can keep this!